test/inductor/test_flex_attention.py

```diff
@@ -593,19 +633,63 @@ def func(q, k, v, k2, v2):
         _, code = run_and_get_code(func, q, k, v, k2, v2)
         # Ensure _create_block_mask_from_mask is compiled and generates 3 kernels,
         # 2 flex_attention generates 2 kernels.
-        FileCheck().check_count(".run(", 5, True).run(code[0])
+        FileCheck().check_count(".run(", 5, False).run(code[0])
 
     @supported_platform
+    @common_utils.parametrize("dtype", test_dtypes_fast)
+    @common_utils.parametrize("k_s", test_input_strides)
+    @common_utils.parametrize("v_s", test_input_strides)
+    @common_utils.parametrize("head_dims", test_Hq_Hkv)
+    def test_strided_decoding(self, dtype: torch.dtype, k_s, v_s, head_dims):
+        Hq, Hkv = head_dims
+        assert Hq % Hkv == 0
+        q1 = torch.randn((B * Hq * D), dtype=dtype, device="cuda")
+        k1 = torch.randn((B * Hkv * S * D * 4), dtype=dtype, device="cuda")
+        v1 = torch.randn((B * Hkv * S * D * 4), dtype=dtype, device="cuda")
+
+        q_shape = (B, H, Hq // Hkv, D)
```

Comment:
It might be more clear if (B, Hkv, Hq // hkv, D)

Reply:
or maybe name some things, group_size

also since users would be expected to do this GQA like grouping, we might want to add a utility for doing this.

Reply:
sorry, my bad. This q_shape isn't used in the code below.(B, Hkv, Hq//Hkv, D) is the shape actually used.